src/cup1d/likelihood/emcee_sampler.py

```diff
@@ -77,645 +119,751 @@ def __init__(self,like=None,
         # Figure out what extra information will be provided as blobs
         self.blobs_dtype = self.like.theory.get_blobs_dtype()
 
-
     def set_truth(self):
-        """ Set up dictionary with true values of cosmological
-        likelihood parameters for plotting purposes """
+        """Set up dictionary with true values of cosmological
+        likelihood parameters for plotting purposes"""
 
         # likelihood contains true parameters, but not in latex names
-        like_truth=self.like.truth
+        like_truth = self.like.truth
 
         # when running on data, we do not know the truth
         if like_truth is None:
-            self.truth=None
+            self.truth = None
             return
 
         # store truth for all parameters, with LaTeX keywords
-        self.truth={}
+        self.truth = {}
         for param in cosmo_params:
-            param_string=param_dict[param]
-            self.truth[param_string]=like_truth[param]
+            param_string = param_dict[param]
+            self.truth[param_string] = like_truth[param]
 
-        return
+        # when using the IGM history of the analyzed simulation set to zero
+        if self.like.data.input_sim == self.like.theory.fid_sim_igm:
+            for param in self.like.free_params[2:]:
```

Comment:
Does this assume that we only have 2 cosmo params? This is the case in most runs, but to test the compression you might want to run with more than 2 parameters (like we did in Pedersen et al. 2023). In general we should be careful not to assume this in cup1d